src/main/java/sparksoniq/jsoniq/runtime/tupleiterator/RuntimeTupleIterator.java

```diff
@@ -104,22 +107,32 @@ public IteratorMetadata getMetadata() {
     public abstract JavaRDD<FlworTuple> getRDD(DynamicContext context);
 
     public abstract boolean isDataFrame();
+    
+    /*
+     * The second parameter gives information on the projection needed by the caller.
+     * Currently, this is used to inform a group by clause that only a count is needed for a specific variable, which allows projecting away the actual items.
+     */
+    public abstract Dataset<Row> getDataFrame(DynamicContext context, Map<String, DynamicContext.VariableDependency> parentProjection);
+
+    /*
+     * Returns the projection needed by this iterator. The result of this method is forwarded to the child clause so it can optimize some values away.
+     */
+    public abstract Map<String, DynamicContext.VariableDependency> getProjection(Map<String, DynamicContext.VariableDependency> parentProjection);
```

Comment:
I really like that these explanations exist.  It can be merged as it is but I have tiny remarks if you'd want to re-iterate over them.

The wording can be more concise and less subject to change (ie. stating '2. parameter' can potentially be misleading after any refactoring involving parameters) . Additionally, javadoc style can be considered.

Reply:
Yes, javadoc is what should be done.